xbmc/cores/VideoPlayer/DVDInputStreams/InputStreamAddon.cpp

```diff
@@ -497,6 +497,11 @@ KODI_HANDLE CInputStreamAddon::cb_get_stream_transfer(KODI_HANDLE handle,
     CDemuxStreamRadioRDS* rdsStream = new CDemuxStreamRadioRDS();
     demuxStream = rdsStream;
   }
+  else if (stream->m_streamType == INPUTSTREAM_TYPE_ID3)
+  {
+    CDemuxStreamAudioID3* id3Stream = new CDemuxStreamAudioID3();
```

Comment:
Not something for this PR, but we prefer std::unique_ptr's. When used correctly all pointer manipulation goes away, e.g.:

```c++
std::unique_ptr<CDemuxStreamAudioID3> id3Stream = std::make_unique<CDemuxStreamAudioID3>();
```

Notice how using the STL class hides all pointer manipulation.

Reply:
Totally agree, I haven't used a raw pointer in any of my personal code in a very long time :) I'm also one of those evil narcissists that forces the issue by making classes implement a static create() method that returns a unique_ptr<> and keeping the real constructor private.  My teammates at work _love_ me for that one, but well, we haven't had a memory leak in like 10 years so .... :)